core/test/scenario/command_handler_test.go

```diff
@@ -0,0 +1,100 @@
+package scenario_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+
+	"github.com/get-eventually/go-eventually/core/aggregate"
+	"github.com/get-eventually/go-eventually/core/command"
+	"github.com/get-eventually/go-eventually/core/event"
+	"github.com/get-eventually/go-eventually/core/internal/user"
+	"github.com/get-eventually/go-eventually/core/test/scenario"
+	"github.com/get-eventually/go-eventually/core/version"
+)
+
+func TestCommandHandler(t *testing.T) {
+	id := uuid.New()
+	now := time.Now()
+
+	t.Run("create new user", func(t *testing.T) {
+		scenario.
+			CommandHandler[user.CreateCommand, user.CreateCommandHandler]().
+			When(command.Envelope[user.CreateCommand]{
```

Comment:
And Go continues to require more boilerplate code than usual :)
I would expect `When` to be able to typecheck `Command` type automatically given that the types are provided.
It feels like it should be possible to structure the DSL a bit differently but idk, I haven't tried using Go generics myself yet.

Reply:
> I would expect When to be able to typecheck Command type automatically given that the types are provided.

Seems like the whole fully-qualified type must be specified when using it in a `type` expression or during `struct` initialization.

Hoping this will be improved in the next releases (similar to what they did with explicit types for elements in a slice/map )